docs/best_practice/systables.txt

```diff
@@ -0,0 +1,437 @@
+
+Crate divides the rows of each table into shards that are distinctively
+distributed to all nodes in your cluster. Replication uses the same
+mechanism to add redundancy and thus resilience to your data. While most
+of the time Crate takes transparently care of distributing and
+replicating the shards, it’s useful during troubleshooting to actually
+find out some more about these data structures. The **sys.shards** table
+provides access to the status and size of shards, its names and IDs::
+
+    cr> SHOW COLUMNS IN shards FROM sys;
+    +--------------------------------+-----------+
+    | column_name                    | data_type |
+    +--------------------------------+-----------+
+    | id                             | integer   |
+    | num_docs                       | long      |
+    | orphan_partition               | boolean   |
+    | partition_ident                | string    |
+    | primary                        | boolean   |
+    | recovery                       | object    |
+    | recovery['files']              | object    |
+    | recovery['files']['percent']   | float     |
+    | recovery['files']['recovered'] | integer   |
+    | recovery['files']['reused']    | integer   |
+    | recovery['files']['used']      | integer   |
+    | recovery['size']               | object    |
+    | recovery['size']['percent']    | float     |
+    | recovery['size']['recovered']  | long      |
+    | recovery['size']['reused']     | long      |
+    | recovery['size']['used']       | long      |
+    | recovery['stage']              | string    |
+    | recovery['total_time']         | long      |
+    | recovery['type']               | string    |
+    | relocating_node                | string    |
+    | routing_state                  | string    |
+    | schema_name                    | string    |
+    | size                           | long      |
+    | state                          | string    |
+    | table_name                     | string    |
+    +--------------------------------+-----------+
+    SHOW 25 rows in set (... sec)
+
+The cluster state is somewhat delicate when nodes join or leave, since
+in those situations shards have to be rearranged to ensure that each of
+them is replicated to different nodes. As long as the **state**
+attribute is ‘STARTED’ for all shards, the cluster is in a stable state,
+otherwise Crate is occupied with some background activity. The cluster
+state indicators on the admin UI evaluate these values as well.
+
+The **sys.shards** table contains even more information about the
+rebalancing activities. Whenever Crate needs to copy one shard to
+another node -- since that may be necessary to ensure there are enough
+replica of it distributed in the cluster -- you can estimate the
+finishing of that operation with the **recovery** object::
+
+    cr> SELECT DISTINCT partition_ident, recovery['size']['percent'], recovery['total_time'] FROM sys.shards WHERE primary;
```

Comment:
This query helps user to estimate how long it takes or will take to recover the shard, for example if 50% is recovered in 1 sec , full recovery will take 2 secs. I think this is what Nils means here.